### PR TITLE
Update trainer logging and sanity checks

### PR DESCRIPTION
**Pull request by nw13slx**
## Description
- let the device be an input argument for init and yaml
- check the consistency between `model.out_fields` and loss function keywords.
- save git commit versions of e3nn, nequip, and torch_runstats

## Motivation and Context
Easier changing of devices. Better safe guard against wrong set up of loss function / model.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] My code follows the code style of this project and has been formatted using `black`.
- [x] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [x] `example.yaml` (and other relevant `configs/`) have been updated with new or changed options.
- [ ] I have updated `CHANGELOG.md`.

### CODE REVIEW

File: nequip/train/trainer.py [hunk 1]
```diff
@@ -213,6 +213,7 @@ def __init__(
         self,
         model,
         model_builders: Optional[list] = [],
+        device: Optional[str] = "cuda" if torch.cuda.is_available() else "cpu",
```
Comment (Linux-cpp-lisp) [inline nequip/train/trainer.py:216]: why `Optional`?

File: nequip/utils/git.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if retcode.returncode == 0:
+        return retcode.stdout.decode().splitlines()[0].split()[0]
+    else:
+        err_info = retcode.stderr.decode()
+        logging.info(err_info)
+        logging.info(
+            f"Fail to retrieve {module} commit version."
+            "It is installe dat {path} but no git is found;"
+            "Try `pip install -e ./` for installation."
+        )
```
Comment (Linux-cpp-lisp) [inline nequip/utils/git.py:28]: this is going to show up in the log of every single install from PyPI... which should be most users eventually

can we remove the log?

File: nequip/utils/git.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        return "NaN"
```
Comment (Linux-cpp-lisp) [inline nequip/utils/git.py:29]: why not just a string that says `"(not `pip install -e`ed from a git repo)"`

File: nequip/train/trainer.py (hunk 1 above)
Comment (nw13slx) [inline nequip/train/trainer.py:216]:
```suggestion
        device: str = "cuda" if torch.cuda.is_available() else "cpu",
```

Comment (nw13slx) [reply nequip/train/trainer.py]: yeah. shouldn't be optional. Removed the optional type

File: nequip/utils/git.py (hunk 2 above)
Comment (nw13slx) [inline nequip/utils/git.py:28]:
```suggestion
```

Comment (nw13slx) [reply nequip/utils/git.py]: removed
